docs/source/installation/guide_complete/index.rst

```diff
@@ -0,0 +1,363 @@
+.. _installation:guide-complete:
+
+===========================
+Complete installation guide
+===========================
+
+The :ref:`quick installation guide <installation:guide-quick>` is designed to make the setup as simple and portable as possible.
+However, the resulting setup has some :ref:`limitations <installation:guide-quick:limitations>` concerning the available functionality and performance.
+This guide provides detailed information and instructions to set up a feature-complete and performant installation.
+
+To get a working installation of AiiDA, the following steps are required:
+
+#. :ref:`Install the Python Package <installation:guide-complete:python-package>`
```

Comment:
I intentionally didn't do this. This can of course be discussed but my thinking is that this should be a generic installation guide. The RabbitMQ service is no longer required for all use cases, and the PostgreSQL service is only required for a particular storage plugin.

If we start with instructions to install PostgreSQL it falsely suggest that this somehow is a requirement for AiiDA while it is merely a requirement for the `core.psql_dos` plugin.

Providing instructions for RMQ already makes more sense since, although optional, many users actually may want to use it. Still, I am struggling to find the best way to add it. There are no straightforward installation instructions that work on all platforms and it often is difficult for users. It don't want them to become completely distracted and make the guide unreadable buy having a million different branches right in the beginning just to install some service. I think it would be better to have these instructions (if we even provide them at all, I don't think that should really be our responsibility). If users cannot install RMQ themselves, they should perhaps just opt for the non-broker solution, which is now available.

Reply:
I can definitely see @sphuber's point of view here. However, maybe we should still add a step for installing the _desired_ services and point to their respective installation instructions. I know @giovannipizzi was still in favor of keeping the instructions we had before.

Also, I don't immediately see the instructions for using `conda` to install and run the services in an environment in e.g. the [RabbitMQ installation docs](https://www.rabbitmq.com/docs/download). This might still be useful to know for several users.

Reply:
Yeah, now that I thought about it again, I'm also on board with that. How about adding something like:

3. [Optional] Use additional services in AiiDA
3. [Optional] Setup AiiDA with additional services

Just to notify the user that these points will follow in the guide? Also phrasing it in that way emphasizes that these are not installation instructions for the services, which we don't provide, but rather how to set things up for AiiDA when they are already installed. We can still add an Info box with the links to the installation instructions on their webpages and the `conda` route, as I mentioned [in my other comment](https://github.com/aiidateam/aiida-core/pull/6455#discussion_r1646136095).

In addition to this, there's a heading for RabbitMQ. We could also add a `# Storage Plugins` heading, and make `core.sqlite_dos` and `core.psql_dos` subsections. Right now, as they are formatted as code, they don't even seem like headings at first glance.

Regarding the instructions for RMQ, we could make the table with the parameters foldable to hide most of that away initially? In principle, I think it's fine to mention RMQ in the way it is done there, no need to go through its installation.

Reply:
> [Optional] Use additional services in AiiDA
> [Optional] Setup AiiDA with additional services

What is the difference between these two points?